task2/InterfaceManager.py

```diff
@@ -0,0 +1,109 @@
+import json
+import logging
+from dataclasses import dataclass, field
+from typing import List
+
+import marshmallow as mm
+import requests
+from dataclasses_json import DataClassJsonMixin, config
+from marshmallow import fields, EXCLUDE
+
+# Logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+formatter = logging.Formatter("%(asctime)s - %(levelname)s - %(message)s")
+
+file_handler = logging.FileHandler('output_folder/interfaces.log', mode='w')
+file_handler.setLevel(logging.INFO)
+file_handler.setFormatter(formatter)
+
+logger.addHandler(file_handler)
+
+
+@dataclass
+class AddressIpv4Inside(DataClassJsonMixin):
+    ip: str
+    netmask: str
+
+
+@dataclass
+class AddressIpv6Inside(DataClassJsonMixin):
+    ip: str
+    prefix_length: int = field(
+        metadata=config(mm_field=mm.fields.Number(data_key='prefix-length')))
+
+
+@dataclass
+class AddressIpv4(DataClassJsonMixin):
+    address: List[AddressIpv4Inside] = field(
+        metadata=config(mm_field=mm.fields.List(fields.Nested(AddressIpv4Inside.schema(unknown=mm.EXCLUDE)))),
+        default_factory=list)
+
+
+@dataclass
+class AddressIpv6(DataClassJsonMixin):
+    address: List[AddressIpv6Inside] = field(
+        metadata=config(mm_field=mm.fields.List(fields.Nested(AddressIpv6Inside.schema(unknown=mm.EXCLUDE)))),
+        default_factory=list)
+
+
+@dataclass
+class Interface(DataClassJsonMixin):
+    name: str
+    description: str = field(metadata=config(mm_field=mm.fields.String(data_key='description', load_default=str)))
+    type: str
+    enabled: bool
+    ip_address_1: AddressIpv4 = field(
+        metadata=config(mm_field=mm.fields.Nested(AddressIpv4.schema(unknown=mm.EXCLUDE), data_key='ietf-ip:ipv4')))
+    ip_address_2: AddressIpv6 = field(
+        metadata=config(mm_field=mm.fields.Nested(AddressIpv6.schema(unknown=mm.EXCLUDE), data_key='ietf-ip:ipv6')))
+
+
+class InterfaceManager:
+    def __init__(self, url="http://127.0.0.1:5000/get-all-interfaces"):
```

Comment:
@patrikprovaznik you shouldn t hold here the whole url.. that is not generic enough for future use. idealy it should contain protocol (https or https), host (ip or hostname) and port (integer) out of which you will construct url in init and keep only that there and all three can be in future configurable from config file